packages/nextra-theme-docs/src/misc/theme.tsx

```diff
@@ -144,87 +131,31 @@ const HeaderLink = ({
   )
 }
 
-interface HeadingProps {
-  children?: React.ReactNode
-  href?: string
-  id: string
-}
-
-const H2 =
-  (context: { index: number }) =>
-  ({ children, ...props }: HeadingProps) => {
-    return (
-      <HeaderLink tag="h2" context={context} {...props}>
-        {children}
-      </HeaderLink>
-    )
-  }
```

Comment:
refactored by adding `createHeaderLink` function to omit code duplication for H2-H6 components